SSR-2360 User guide for Custom CSS (#1134)
kintone-labs/kintone-ui-component: docs/document/docs/getting-started/custom-css.md

---
id: custom-css
title: Custom CSS
sidebar_label: Custom CSS
---

## Overview

> **Kintone UI Component** provide some Custom CSS properties for each component by using [CSS inheritance](#css-inheritance) and [CSS variables and custom properties](#css-variables-and-custom-properties) together, it's easy to create themable elements. By applying css selectors to customize CSS custom properties, tree-based and per-instance theming is straightforward to apply.

Author: themable element meaning an element that we can customize its css.
I will change "themable" to "customizable".

> **Kintone UI Component** provide some Custom CSS properties for each component by using [CSS inheritance](#css-inheritance) and [CSS variables and custom properties](#css-variables-and-custom-properties) together, it's easy to create themable elements. By applying css selectors to customize CSS custom properties, tree-based and per-instance theming is straightforward to apply.

Author: I also change to v1.5.0.
Thank you

> **Kintone UI Component** provide some Custom CSS properties for each component by using [CSS inheritance](#css-inheritance) and [CSS variables and custom properties](#css-variables-and-custom-properties) together, it's easy to create themable elements. By applying css selectors to customize CSS custom properties, tree-based and per-instance theming is straightforward to apply.

Reviewer: Got it, thank you!

> **Kintone UI Component (KUC)** provides some Custom CSS properties for each component by using [CSS inheritance](#css-inheritance) and [CSS variables and custom properties](#css-variables-and-custom-properties) together. It's easy to create customizable elements. By applying CSS selectors to customize CSS custom properties, tree-based and per-instance theming are straightforward to apply.

Reviewer 2: ```suggestion
**Kintone UI Component (KUC)** provides some Custom CSS properties for each component by using [CSS inheritance](#css-inheritance), [CSS variables and custom properties](#css-variables-and-custom-properties) together. It's easy to create customizable elements by applying CSS selectors to customize CSS custom properties. Tree-based and per-instance theming are straightforward to apply.
```

### CSS inheritance

CSS inheritance lets parent and host elements propagate certain CSS properties to their descendants.

### CSS variables and custom properties

> All CSS custom properties (`--custom-property-name`) inherit. You can use this to make your component's styles configurable from outside.

Reviewer 2: ```suggestion
All CSS custom properties (`--custom-property-name`) inherit their value from their parent. You can use this to make your component's styles configurable from the outside.
```
How about this to make the sentence clearer?

> KUC will publish some properties for each element, based on that users can customize CSS easily.

Reviewer 2: ```suggestion
KUC will publish some properties for each element and based on that, users can easily customize CSS properties of the element.
```

> KUC will publicize some properties for each element, based on that users can customize css easily (_For now we will publicize some properties for **Button** and **Text** component first_).

Author: >What is the meaning of "publicize" in this case?
I change "publicize" to "publish".

>I think it is good not to mention the component names.
Ok, I remove it

## Usage

> You can set the value for `--my-background` by using the CSS element selector:

Reviewer 2: ```suggestion
You can set the value for `--my-background` by using the CSS selectors:
```
It seems like you can use not only element selectors but other CSS selectors in general.

```javascript
<style>
  my-element {
    --my-background: #d9092f;
  }
</style>
<my-element></my-element>
```

`--my-background` is configurable per instance of `my-element`:

```javascript
<style>
  my-element {
    --my-background: #d9092f;
  }
  my-element.sample-class {
    --my-background: #a8a632;
  }
   my-element#sample-id {
    --my-background: #1ac9b8;

>   }

Reviewer: You are using "my-element" before class name/id here.
And on Sample Code of Text, you are directly using class name/id as element selector.
I think it's better to unify it.

</style>
<my-element></my-element>
<my-element class="sample-class"></my-element>
<my-element id="sample-id"></my-element>
```

> > You can find the list of Custom CSS properties in the **Custom CSS** section on each component page.

Reviewer 2: ```suggestion
> You can find a list of the Custom CSS properties in the **Custom CSS** section on each component page.
```
